editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/utils/MavenUtils.java

```diff
@@ -66,14 +68,22 @@ public String javaSourceFolder() throws CoreException {
 	 * @param compDeps the Maven dependencies to be updated
 	 * @throws CoreException
      */
-	public void updateMavenDependencies(List<org.fusesource.ide.camel.model.service.core.catalog.Dependency> compDeps) throws CoreException {
-        IProject project = CamelUtils.project();
+	public void updateMavenDependencies(final List<org.fusesource.ide.camel.model.service.core.catalog.Dependency> compDeps) throws CoreException {
+        final IProject project = CamelUtils.project();
         if (project == null) {
             CamelEditorUIActivator.pluginLog().logWarning("Unable to add component dependencies because selected project can't be determined. Maybe this is a remote camel context.");
             return;
         }
-
-        updateMavenDependencies(compDeps, project);
+        Display.getDefault().asyncExec(new Runnable() {
```

Comment:
if it is async, we might need to wait for it to complete, otherwise the properties view won't have the correct content, no?